### PR TITLE
Correct SA1629 to add a period if the documentation ends in an xml entity. Previously replaced the semicolon in the xml entity.

### LINKED ISSUE #3802: `SA1629` eats ending semicolon
``` csharp
/// <summary>
/// Indicates that this property is a dictionary containing SAML attribute values.
/// </summary>
/// <remarks>This attribute should only be placed on a Dictionary&lt;string, string?&gt;</remarks>
[AttributeUsage(AttributeTargets.Property)]
public sealed class SamlAttributesAttribute : System.Attribute;
```

The fixer for `SA1629` will replace the `;` with `.` instead of appending it, thereby breaking the XML syntax.

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #3802.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs [hunk 1]
```diff
@@ -131,8 +134,9 @@ private static void HandleSectionOrBlockXmlElement(SyntaxNodeAnalysisContext con
                             {
                                 int spanStart = textToken.SpanStart + textWithoutTrailingWhitespace.Length;
                                 ImmutableDictionary<string, string> properties = null;
-                                if (textWithoutTrailingWhitespace.EndsWith(",", StringComparison.Ordinal)
-                                    || textWithoutTrailingWhitespace.EndsWith(";", StringComparison.Ordinal))
+                                if (textWithoutTrailingWhitespace.EndsWith(",", StringComparison.Ordinal) ||
+                                    (textWithoutTrailingWhitespace.EndsWith(";", StringComparison.Ordinal) &&
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs:138]: 💡 Keep `||` and `&&` at the beginning of the line

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs [hunk 2]
```diff
@@ -131,8 +134,9 @@ private static void HandleSectionOrBlockXmlElement(SyntaxNodeAnalysisContext con
...
+                                     !XmlEntityRegex.IsMatch(textWithoutTrailingWhitespace)))
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs:139]: 💭 Can we not use `textToken` here?

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs (hunk 1 above)
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs]: Reverted

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs (hunk 2 above)
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs]: Updated to check syntax kind instead

**Review by sharwell — APPROVED** [review]

**Pull request merged**